homeassistant/components/websocket_api/commands.py

```diff
@@ -127,54 +131,84 @@ def handle_unsubscribe_events(hass, connection, msg):
 @decorators.async_response
 async def handle_call_service(hass, connection, msg):
     """Handle call service command."""
-    blocking = True
-    if msg["domain"] == HASS_DOMAIN and msg["service"] in ["restart", "stop"]:
-        blocking = False
-
     # We do not support templates.
     target = msg.get("target")
     if template.is_complex(target):
         raise vol.Invalid("Templates are not supported here")
 
+    def _prepare_service_data():
+        return msg["domain"], msg["service"], msg.get("service_data"), target
+
+    await _handle_call_service(hass, connection, msg["id"], _prepare_service_data)
+
+
+@decorators.websocket_command(
+    {
+        vol.Required("type"): "call_service_action",
+        vol.Required("service_action"): dict,
```

Comment:
This should be a list and validate with `cv.SCRIPT_SCHEMA`

Reply:
Why should it be a list is it for same reason as the comment on line 147?
Validation is done below:
```py
        config = async_prepare_call_from_config(
            hass, msg["service_action"], validate_config=True
        )
```